MGARD Fails to Build with Clang

I am trying to build @pnorbert's ADIOS2 from the latest Spack v0.21.1.

This seems to by default not activate mgard (cc @robertu94), which in turn fails to build:
```
2024-02-26T23:41:26.0081405Z ==> Installing mgard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su [29/30]
2024-02-26T23:41:26.0082434Z ==> No binary for mgard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su found: installing from source
2024-02-26T23:41:28.6387571Z ==> No patches needed for mgard
2024-02-26T23:41:28.6465506Z ==> mgard: Executing phase: 'cmake'
2024-02-26T23:41:29.3497402Z ==> mgard: Executing phase: 'build'
2024-02-26T23:46:42.0273319Z ==> Error: ProcessError: Command exited with status 2:
2024-02-26T23:46:42.0274231Z     'make' '-j2'
2024-02-26T23:46:42.4126820Z 
2024-02-26T23:46:42.4127487Z 15 errors found in build log:
2024-02-26T23:46:42.4128700Z      805    make[2]: Entering directory '/tmp/runner/spack-stage/spack-stage-mg
2024-02-26T23:46:42.4129575Z             ard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-build-jdfekcq
2024-02-26T23:46:42.4130560Z             '
2024-02-26T23:46:42.4131080Z      806    [ 99%] Building CXX object CMakeFiles/mgard-x.dir/src/mgard-x/Execu
2024-02-26T23:46:42.4131702Z             tables/mgard-x.cpp.o
2024-02-26T23:46:42.4132284Z      807    /opt/spack/lib/spack/env/clang/clang++ -DMGARD_ENABLE_OPENMP -DMGAR
2024-02-26T23:46:42.4133189Z             D_ENABLE_SERIAL -DMGARD_ZSTD -I/tmp/runner/spack-stage/spack-stage-
2024-02-26T23:46:42.4133983Z             mgard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/include
2024-02-26T23:46:42.4134742Z              -I/tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgps
2024-02-26T23:46:42.4135511Z             cmer6ywj2mqd6ottf77su/spack-build-jdfekcq/include -isystem /opt/spa
2024-02-26T23:46:42.4136254Z             ck/opt/spack/linux-ubuntu20.04-x86_64/clang-7.0.1/protobuf-3.21.12-
2024-02-26T23:46:42.4136983Z             kfkqr74bmboadakntrrbu224st2yfok3/include -isystem /opt/spack/opt/sp
2024-02-26T23:46:42.4137733Z             ack/linux-ubuntu20.04-x86_64/clang-7.0.1/zlib-ng-2.1.4-sitpgf77f4vk
2024-02-26T23:46:42.4138459Z             u6fmztnl5zumieypphgh/include -isystem /opt/spack/opt/spack/linux-ub
2024-02-26T23:46:42.4139178Z             untu20.04-x86_64/clang-7.0.1/zstd-1.5.5-x6qi7cjbpsgzhl5uxyrewc2mata
2024-02-26T23:46:42.4139895Z             ugrau/include -w -O3 -DNDEBUG -std=gnu++17 -fopenmp=libomp -fopenmp
2024-02-26T23:46:42.4140599Z             =libomp -MD -MT CMakeFiles/mgard-x.dir/src/mgard-x/Executables/mgar
2024-02-26T23:46:42.4141269Z             d-x.cpp.o -MF CMakeFiles/mgard-x.dir/src/mgard-x/Executables/mgard-
2024-02-26T23:46:42.4142015Z             x.cpp.o.d -o CMakeFiles/mgard-x.dir/src/mgard-x/Executables/mgard-x
2024-02-26T23:46:42.4142743Z             .cpp.o -c /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfe
2024-02-26T23:46:42.4143471Z             kcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgar
2024-02-26T23:46:42.4144034Z             d-x.cpp
2024-02-26T23:46:42.4144549Z      808    [ 99%] Building CXX object CMakeFiles/mgard-x-autotuner.dir/src/mga
2024-02-26T23:46:42.4145180Z             rd-x/Executables/mgard-x-autotuner.cpp.o
2024-02-26T23:46:42.4146124Z      809    /opt/spack/lib/spack/env/clang/clang++ -DMGARD_ENABLE_OPENMP -DMGAR
2024-02-26T23:46:42.4146886Z             D_ENABLE_SERIAL -DMGARD_ZSTD -I/tmp/runner/spack-stage/spack-stage-
2024-02-26T23:46:42.4147661Z             mgard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/include
2024-02-26T23:46:42.4148431Z              -I/tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgps
2024-02-26T23:46:42.4149134Z             cmer6ywj2mqd6ottf77su/spack-build-jdfekcq/include -isystem /opt/spa
2024-02-26T23:46:42.4149880Z             ck/opt/spack/linux-ubuntu20.04-x86_64/clang-7.0.1/protobuf-3.21.12-
2024-02-26T23:46:42.4150619Z             kfkqr74bmboadakntrrbu224st2yfok3/include -isystem /opt/spack/opt/sp
2024-02-26T23:46:42.4151338Z             ack/linux-ubuntu20.04-x86_64/clang-7.0.1/zlib-ng-2.1.4-sitpgf77f4vk
2024-02-26T23:46:42.4152079Z             u6fmztnl5zumieypphgh/include -isystem /opt/spack/opt/spack/linux-ub
2024-02-26T23:46:42.4152793Z             untu20.04-x86_64/clang-7.0.1/zstd-1.5.5-x6qi7cjbpsgzhl5uxyrewc2mata
2024-02-26T23:46:42.4153470Z             ugrau/include -w -O3 -DNDEBUG -std=gnu++17 -fopenmp=libomp -fopenmp
2024-02-26T23:46:42.4154209Z             =libomp -MD -MT CMakeFiles/mgard-x-autotuner.dir/src/mgard-x/Execut
2024-02-26T23:46:42.4154933Z             ables/mgard-x-autotuner.cpp.o -MF CMakeFiles/mgard-x-autotuner.dir/
2024-02-26T23:46:42.4155674Z             src/mgard-x/Executables/mgard-x-autotuner.cpp.o.d -o CMakeFiles/mga
2024-02-26T23:46:42.4156461Z             rd-x-autotuner.dir/src/mgard-x/Executables/mgard-x-autotuner.cpp.o 
2024-02-26T23:46:42.4157188Z             -c /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgps
2024-02-26T23:46:42.4157913Z             cmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x-aut
2024-02-26T23:46:42.4158455Z             otuner.cpp
2024-02-26T23:46:42.4159122Z      810    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4159914Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4160508Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4161086Z   >> 811    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4161935Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4162662Z             tor.h:45:12: error: no member named 'sqrt' in namespace 'std'; did 
2024-02-26T23:46:42.4163220Z             you mean simply 'sqrt'?
2024-02-26T23:46:42.4175088Z      812        return std::sqrt(L_2);
2024-02-26T23:46:42.4175501Z      813               ^~~~~~~~~
2024-02-26T23:46:42.4175862Z      814               sqrt
2024-02-26T23:46:42.4176520Z      815    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1038:1: note: 'sqrt' d
2024-02-26T23:46:42.4177063Z             eclared here
2024-02-26T23:46:42.4177572Z      816    sqrt(_A1 __lcpp_x) _NOEXCEPT {return ::sqrt((double)__lcpp_x);}
2024-02-26T23:46:42.4178073Z      817    ^
2024-02-26T23:46:42.4178598Z      818    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4179373Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4179999Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4180624Z   >> 819    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4181394Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4182092Z             tor.h:47:12: error: no member named 'sqrt' in namespace 'std'; did 
2024-02-26T23:46:42.4182659Z             you mean simply 'sqrt'?
2024-02-26T23:46:42.4183076Z      820        return std::sqrt(L_2 / n);
2024-02-26T23:46:42.4183473Z      821               ^~~~~~~~~
2024-02-26T23:46:42.4183820Z      822               sqrt
2024-02-26T23:46:42.4184369Z      823    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1038:1: note: 'sqrt' d
2024-02-26T23:46:42.4184882Z             eclared here
2024-02-26T23:46:42.4185533Z      824    sqrt(_A1 __lcpp_x) _NOEXCEPT {return ::sqrt((double)__lcpp_x);}
2024-02-26T23:46:42.4186070Z      825    ^
2024-02-26T23:46:42.4186611Z      826    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4187353Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4187968Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4188573Z   >> 827    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4189308Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4190045Z             tor.h:115:15: error: no member named 'log10' in namespace 'std'; di
2024-02-26T23:46:42.4190610Z             d you mean simply 'log10'?
2024-02-26T23:46:42.4191053Z      828      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4191538Z      829                  ^~~~~~~~~~
2024-02-26T23:46:42.4191899Z      830                  log10
2024-02-26T23:46:42.4192477Z      831    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:972:1: note: 'log10' d
2024-02-26T23:46:42.4192996Z             eclared here
2024-02-26T23:46:42.4193429Z      832    log10(_A1 __lcpp_x) _NOEXCEPT {return ::log10((double)__lcpp_x);}
2024-02-26T23:46:42.4193929Z      833    ^
2024-02-26T23:46:42.4194441Z      834    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4195198Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4195813Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4196396Z   >> 835    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4197133Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4198039Z             tor.h:115:34: error: no member named 'sqrt' in namespace 'std'; did
2024-02-26T23:46:42.4198590Z              you mean simply 'sqrt'?
2024-02-26T23:46:42.4199056Z      836      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4199566Z      837                                     ^~~~~~~~~
2024-02-26T23:46:42.4199993Z      838                                     sqrt
2024-02-26T23:46:42.4200699Z      839    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1038:1: note: 'sqrt' d
2024-02-26T23:46:42.4201256Z             eclared here
2024-02-26T23:46:42.4201716Z      840    sqrt(_A1 __lcpp_x) _NOEXCEPT {return ::sqrt((double)__lcpp_x);}
2024-02-26T23:46:42.4202221Z      841    ^
2024-02-26T23:46:42.4202711Z      842    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4203475Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4204082Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4204698Z   >> 843    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4205440Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4206138Z             tor.h:45:17: error: call to function 'sqrt' that is neither visible
2024-02-26T23:46:42.4206853Z              in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4207387Z      844        return std::sqrt(L_2);
2024-02-26T23:46:42.4207777Z      845                    ^
2024-02-26T23:46:42.4208348Z      846    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4209081Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4209716Z             tor.h:81:16: note: in instantiation of function template specializa
2024-02-26T23:46:42.4210425Z             tion 'mgard_x::L_2_norm<double>' requested here
2024-02-26T23:46:42.4211019Z      847      T org_norm = L_2_norm<T>(shape, original_data, normalize_coordina
2024-02-26T23:46:42.4211551Z             tes);
2024-02-26T23:46:42.4211846Z      848                   ^
2024-02-26T23:46:42.4212389Z      849    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4213154Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:25
2024-02-26T23:46:42.4213836Z             3:29: note: in instantiation of function template specialization 'm
2024-02-26T23:46:42.4214474Z             gard_x::L_2_error<double>' requested here
2024-02-26T23:46:42.4214764Z 
2024-02-26T23:46:42.4214889Z      ...
2024-02-26T23:46:42.4215030Z 
2024-02-26T23:46:42.4215192Z      856          launch_compress<double>(
2024-02-26T23:46:42.4215569Z      857          ^
2024-02-26T23:46:42.4216089Z      858    /usr/include/x86_64-linux-gnu/bits/mathcalls.h:143:13: note: 'sqrt'
2024-02-26T23:46:42.4216669Z              should be declared prior to the call site
2024-02-26T23:46:42.4217162Z      859    __MATHCALL (sqrt,, (_Mdouble_ __x));
2024-02-26T23:46:42.4217575Z      860                ^
2024-02-26T23:46:42.4218090Z      861    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4218869Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4219458Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4220071Z   >> 862    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4220828Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4221508Z             tor.h:47:17: error: call to function 'sqrt' that is neither visible
2024-02-26T23:46:42.4222216Z              in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4222770Z      863        return std::sqrt(L_2 / n);
2024-02-26T23:46:42.4223162Z      864                    ^
2024-02-26T23:46:42.4223722Z      865    /usr/include/x86_64-linux-gnu/bits/mathcalls.h:143:13: note: 'sqrt'
2024-02-26T23:46:42.4224381Z              should be declared prior to the call site
2024-02-26T23:46:42.4224831Z      866    __MATHCALL (sqrt,, (_Mdouble_ __x));
2024-02-26T23:46:42.4225368Z      867                ^
2024-02-26T23:46:42.4226037Z      868    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4226909Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4227556Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4228151Z   >> 869    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4228894Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4229622Z             tor.h:115:39: error: call to function 'sqrt' that is neither visibl
2024-02-26T23:46:42.4230325Z             e in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4230921Z      870      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4231423Z      871                                          ^
2024-02-26T23:46:42.4232030Z      872    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4232757Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:27
2024-02-26T23:46:42.4233452Z             7:37: note: in instantiation of function template specialization 'm
2024-02-26T23:46:42.4234046Z             gard_x::PSNR<double>' requested here
2024-02-26T23:46:42.4234574Z      873                << "PSNR: " << mgard_x::PSNR(n, original_data, decompre
2024-02-26T23:46:42.4235004Z             ssed_data)
2024-02-26T23:46:42.4235340Z      874                                        ^
2024-02-26T23:46:42.4235989Z      875    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4236734Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:39
2024-02-26T23:46:42.4237446Z             3:5: note: in instantiation of function template specialization 'pr
2024-02-26T23:46:42.4238069Z             int_statistics<double>' requested here
2024-02-26T23:46:42.4238352Z 
2024-02-26T23:46:42.4238452Z      ...
2024-02-26T23:46:42.4238617Z 
2024-02-26T23:46:42.4238761Z      879          launch_compress<double>(
2024-02-26T23:46:42.4239151Z      880          ^
2024-02-26T23:46:42.4239670Z      881    /usr/include/x86_64-linux-gnu/bits/mathcalls.h:143:13: note: 'sqrt'
2024-02-26T23:46:42.4240248Z              should be declared prior to the call site
2024-02-26T23:46:42.4240705Z      882    __MATHCALL (sqrt,, (_Mdouble_ __x));
2024-02-26T23:46:42.4241104Z      883                ^
2024-02-26T23:46:42.4241648Z      884    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4242422Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4243019Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4243595Z   >> 885    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4244353Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4245066Z             tor.h:115:20: error: call to function 'log10' that is neither visib
2024-02-26T23:46:42.4245777Z             le in the template definition nor found by argument-dependent looku
2024-02-26T23:46:42.4246324Z             p
2024-02-26T23:46:42.4246709Z      886      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4247151Z      887                       ^
2024-02-26T23:46:42.4247728Z      888    /usr/include/x86_64-linux-gnu/bits/mathcalls.h:107:13: note: 'log10
2024-02-26T23:46:42.4248351Z             ' should be declared prior to the call site
2024-02-26T23:46:42.4248834Z      889    __MATHCALL (log10,, (_Mdouble_ __x));
2024-02-26T23:46:42.4249241Z      890                ^
2024-02-26T23:46:42.4249749Z      891    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4250621Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4251342Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4251957Z   >> 892    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4252735Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4253487Z             tor.h:45:17: error: call to function 'sqrt' that is neither visible
2024-02-26T23:46:42.4254212Z              in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4435940Z      893        return std::sqrt(L_2);
2024-02-26T23:46:42.4436293Z      894                    ^
2024-02-26T23:46:42.4437305Z      895    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4438257Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4438908Z             tor.h:81:16: note: in instantiation of function template specializa
2024-02-26T23:46:42.4439578Z             tion 'mgard_x::L_2_norm<float>' requested here
2024-02-26T23:46:42.4440089Z      896      T org_norm = L_2_norm<T>(shape, original_data, normalize_coordina
2024-02-26T23:46:42.4440527Z             tes);
2024-02-26T23:46:42.4440761Z      897                   ^
2024-02-26T23:46:42.4441278Z      898    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4441970Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:25
2024-02-26T23:46:42.4442599Z             3:29: note: in instantiation of function template specialization 'm
2024-02-26T23:46:42.4443150Z             gard_x::L_2_error<float>' requested here
2024-02-26T23:46:42.4443418Z 
2024-02-26T23:46:42.4443504Z      ...
2024-02-26T23:46:42.4443624Z 
2024-02-26T23:46:42.4443744Z      905          launch_compress<float>(
2024-02-26T23:46:42.4444054Z      906          ^
2024-02-26T23:46:42.4444514Z      907    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1031:46: note: 'sqrt' 
2024-02-26T23:46:42.4445030Z             should be declared prior to the call site
2024-02-26T23:46:42.4445544Z      908    inline _LIBCPP_INLINE_VISIBILITY float       sqrt(float __lcpp_x) _
2024-02-26T23:46:42.4446056Z             NOEXCEPT       {return ::sqrtf(__lcpp_x);}
2024-02-26T23:46:42.4446447Z      909                                                 ^
2024-02-26T23:46:42.4447012Z      910    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4447712Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4448244Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4448833Z   >> 911    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4449522Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4450162Z             tor.h:47:17: error: call to function 'sqrt' that is neither visible
2024-02-26T23:46:42.4450810Z              in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4451285Z      912        return std::sqrt(L_2 / n);
2024-02-26T23:46:42.4451610Z      913                    ^
2024-02-26T23:46:42.4452095Z      914    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1031:46: note: 'sqrt' 
2024-02-26T23:46:42.4452592Z             should be declared prior to the call site
2024-02-26T23:46:42.4453101Z      915    inline _LIBCPP_INLINE_VISIBILITY float       sqrt(float __lcpp_x) _
2024-02-26T23:46:42.4453607Z             NOEXCEPT       {return ::sqrtf(__lcpp_x);}
2024-02-26T23:46:42.4453999Z      916                                                 ^
2024-02-26T23:46:42.4454558Z      917    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4455251Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4455778Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4456310Z   >> 918    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4457230Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4457876Z             tor.h:115:39: error: call to function 'sqrt' that is neither visibl
2024-02-26T23:46:42.4458526Z             e in the template definition nor found by argument-dependent lookup
2024-02-26T23:46:42.4459145Z      919      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4459560Z      920                                          ^
2024-02-26T23:46:42.4460131Z      921    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4460811Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:27
2024-02-26T23:46:42.4461425Z             7:37: note: in instantiation of function template specialization 'm
2024-02-26T23:46:42.4461953Z             gard_x::PSNR<float>' requested here
2024-02-26T23:46:42.4462382Z      922                << "PSNR: " << mgard_x::PSNR(n, original_data, decompre
2024-02-26T23:46:42.4462779Z             ssed_data)
2024-02-26T23:46:42.4463055Z      923                                        ^
2024-02-26T23:46:42.4463615Z      924    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4464292Z             r6ywj2mqd6ottf77su/spack-src/src/mgard-x/Executables/mgard-x.cpp:39
2024-02-26T23:46:42.4464938Z             3:5: note: in instantiation of function template specialization 'pr
2024-02-26T23:46:42.4465727Z             int_statistics<float>' requested here
2024-02-26T23:46:42.4466001Z 
2024-02-26T23:46:42.4466082Z      ...
2024-02-26T23:46:42.4466210Z 
2024-02-26T23:46:42.4466321Z      928          launch_compress<float>(
2024-02-26T23:46:42.4466636Z      929          ^
2024-02-26T23:46:42.4467088Z      930    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:1031:46: note: 'sqrt' 
2024-02-26T23:46:42.4467592Z             should be declared prior to the call site
2024-02-26T23:46:42.4468101Z      931    inline _LIBCPP_INLINE_VISIBILITY float       sqrt(float __lcpp_x) _
2024-02-26T23:46:42.4468614Z             NOEXCEPT       {return ::sqrtf(__lcpp_x);}
2024-02-26T23:46:42.4469012Z      932                                                 ^
2024-02-26T23:46:42.4469577Z      933    In file included from /tmp/runner/spack-stage/spack-stage-mgard-202
2024-02-26T23:46:42.4470267Z             3-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-src/src/mgard-x/Exec
2024-02-26T23:46:42.4470799Z             utables/mgard-x.cpp:17:
2024-02-26T23:46:42.4471338Z   >> 934    /tmp/runner/spack-stage/spack-stage-mgard-2023-03-31-jdfekcqcgpscme
2024-02-26T23:46:42.4472013Z             r6ywj2mqd6ottf77su/spack-src/include/mgard-x/Utilities/ErrorCalcula
2024-02-26T23:46:42.4472654Z             tor.h:115:20: error: call to function 'log10' that is neither visib
2024-02-26T23:46:42.4473290Z             le in the template definition nor found by argument-dependent looku
2024-02-26T23:46:42.4473753Z             p
2024-02-26T23:46:42.4474074Z      935      return 20 * std::log10(range / std::sqrt(mse));
2024-02-26T23:46:42.4474458Z      936                       ^
2024-02-26T23:46:42.4474955Z      937    /usr/lib/llvm-7/bin/../include/c++/v1/math.h:965:46: note: 'log10' 
2024-02-26T23:46:42.4475464Z             should be declared prior to the call site
2024-02-26T23:46:42.4475964Z      938    inline _LIBCPP_INLINE_VISIBILITY float       log10(float __lcpp_x) 
2024-02-26T23:46:42.4476470Z             _NOEXCEPT       {return ::log10f(__lcpp_x);}
2024-02-26T23:46:42.4476867Z      939                                                 ^
2024-02-26T23:46:42.4477223Z      940    12 errors generated.
2024-02-26T23:46:42.4477704Z   >> 941    make[2]: *** [CMakeFiles/mgard-x.dir/build.make:79: CMakeFiles/mgar
2024-02-26T23:46:42.4478289Z             d-x.dir/src/mgard-x/Executables/mgard-x.cpp.o] Error 1
2024-02-26T23:46:42.4478896Z      942    make[2]: Leaving directory '/tmp/runner/spack-stage/spack-stage-mga
2024-02-26T23:46:42.4479593Z             rd-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-build-jdfekcq'
2024-02-26T23:46:42.4480391Z   >> 943    make[1]: *** [CMakeFiles/Makefile2:884: CMakeFiles/mgard-x.dir/all]
2024-02-26T23:46:42.4480815Z              Error 2
2024-02-26T23:46:42.4481122Z      944    make[1]: *** Waiting for unfinished jobs....
2024-02-26T23:46:42.4481623Z      945    [100%] Linking CXX executable bin/mgard-x-autotuner
2024-02-26T23:46:42.4482326Z      946    /usr/local/bin/cmake -E cmake_link_script CMakeFiles/mgard-x-autotu
2024-02-26T23:46:42.4482860Z             ner.dir/link.txt --verbose=1
2024-02-26T23:46:42.4483414Z      947    /opt/spack/lib/spack/env/clang/clang++  -w -O3 -DNDEBUG "CMakeFiles
2024-02-26T23:46:42.4484104Z             /mgard-x-autotuner.dir/src/mgard-x/Executables/mgard-x-autotuner.cp
2024-02-26T23:46:42.4484763Z             p.o" -o bin/mgard-x-autotuner  -Wl,-rpath,/tmp/runner/spack-stage/s
2024-02-26T23:46:42.4485450Z             pack-stage-mgard-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-
2024-02-26T23:46:42.4486162Z             build-jdfekcq/lib:/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/cla
2024-02-26T23:46:42.4486855Z             ng-7.0.1/protobuf-3.21.12-kfkqr74bmboadakntrrbu224st2yfok3/lib:/opt
2024-02-26T23:46:42.4487524Z             /spack/opt/spack/linux-ubuntu20.04-x86_64/clang-7.0.1/zlib-ng-2.1.4
2024-02-26T23:46:42.4488185Z             -sitpgf77f4vku6fmztnl5zumieypphgh/lib:/opt/spack/opt/spack/linux-ub
2024-02-26T23:46:42.4488835Z             untu20.04-x86_64/clang-7.0.1/zstd-1.5.5-x6qi7cjbpsgzhl5uxyrewc2mata
2024-02-26T23:46:42.4489367Z             ugrau/lib::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4489809Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4490202Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4490592Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4490986Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4491372Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4491756Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4492146Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4492537Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4492920Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4493302Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4493684Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4494062Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4494448Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4494837Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4495226Z             :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
2024-02-26T23:46:42.4495710Z             ::::::::::::::::::::::::: lib/libmgard.so.1.5.0 -ldl /opt/spack/opt
2024-02-26T23:46:42.4496289Z             /spack/linux-ubuntu20.04-x86_64/clang-7.0.1/protobuf-3.21.12-kfkqr7
2024-02-26T23:46:42.4496871Z             4bmboadakntrrbu224st2yfok3/lib/libprotobuf.so /opt/spack/opt/spack/
2024-02-26T23:46:42.4497519Z             linux-ubuntu20.04-x86_64/clang-7.0.1/zlib-ng-2.1.4-sitpgf77f4vku6fm
2024-02-26T23:46:42.4498163Z             ztnl5zumieypphgh/lib/libz.so /opt/spack/opt/spack/linux-ubuntu20.04
2024-02-26T23:46:42.4498819Z             -x86_64/clang-7.0.1/zstd-1.5.5-x6qi7cjbpsgzhl5uxyrewc2mataugrau/lib
2024-02-26T23:46:42.4499455Z             /libzstd.so /usr/lib/llvm-7/lib/libomp.so /usr/lib/x86_64-linux-gnu
2024-02-26T23:46:42.4499964Z             /libpthread.so
2024-02-26T23:46:42.4500438Z      948    make[2]: Leaving directory '/tmp/runner/spack-stage/spack-stage-mga
2024-02-26T23:46:42.4501109Z             rd-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-build-jdfekcq'
2024-02-26T23:46:42.4501670Z      949    [100%] Built target mgard-x-autotuner
2024-02-26T23:46:42.4502273Z      950    make[1]: Leaving directory '/tmp/runner/spack-stage/spack-stage-mga
2024-02-26T23:46:42.4502944Z             rd-2023-03-31-jdfekcqcgpscmer6ywj2mqd6ottf77su/spack-build-jdfekcq'
2024-02-26T23:46:42.4503445Z   >> 951    make: *** [Makefile:139: all] Error 2
```

--- a/include/mgard-x/Utilities/ErrorCalculator.h
+++ b/include/mgard-x/Utilities/ErrorCalculator.h
@@ -13,6 +13,11 @@
 // #include "../../shuffle.hpp"
 #include "Types.h"
 
+#include <cmath>
+#include <limits>
+#include <vector>
+
+
 namespace mgard_x {
 
 template <typename T> T L_inf_norm(size_t n, const T *data) {

